test/hipgraph/test_hipgraph_algs.cpp

```diff
@@ -0,0 +1,229 @@
+    const seed_type seed_value = seeds[random_seeds_count - 1];
+    SCOPED_TRACE(testing::Message() << "with seed = " << seed_value);
+    SCOPED_TRACE(testing::Message() << "with sort_data_size = " << sort_data_size);
+    SCOPED_TRACE(testing::Message() << "with search_needle_size = " << search_needle_size);
+
+    // Allocate device buffers and copy data into them
+    key_type* d_sort_input = nullptr;
+    key_type* d_sort_output = nullptr; // also used as search_input
+    key_type* d_search_output = nullptr;
+    key_type* d_search_needles = nullptr;
+    HIP_CHECK(test_common_utils::hipMallocHelper(&d_sort_input, sort_data_size * sizeof(key_type)));
+    HIP_CHECK(test_common_utils::hipMallocHelper(&d_sort_output, sort_data_size * sizeof(key_type)));
+    HIP_CHECK(test_common_utils::hipMallocHelper(&d_search_output, search_needle_size * sizeof(key_type)));
+    HIP_CHECK(test_common_utils::hipMallocHelper(&d_search_needles, search_needle_size * sizeof(key_type)));
+
+    // Default stream does not support hipGraph stream capture, so create a non-blocking one
+    hipStream_t stream = 0;
+    HIP_CHECK(hipStreamCreateWithFlags(&stream, hipStreamNonBlocking));
+
+    // Begin graph capture
+    hipGraph_t graph = test_utils::createGraphHelper(stream);
+
+    // Get temporary storage size required for merge_sort.
+    // Note: doing this insize a graph doesn't gain us any benefit,
+    // since these calls run entirely on the host - however, it is
+    // important to validate that they work inside a graph capture block.
+    size_t sort_temp_storage_bytes = 0;
+    HIP_CHECK(rocprim::merge_sort(nullptr,
+                                  sort_temp_storage_bytes,
+                                  d_sort_input,
+                                  d_sort_output,
+                                  sort_data_size,
+                                  compare_op,
+                                  stream,
+                                  debug_synchronous
+                                  ));
+
+    // Get size of temporary storage required for binary_search
+    size_t search_temp_storage_bytes = 0;
+    HIP_CHECK(rocprim::binary_search(nullptr,
+                                     search_temp_storage_bytes,
+                                     d_sort_output,
+                                     d_search_needles,
+                                     d_search_output,
+                                     sort_data_size,
+                                     search_needle_size,
+                                     compare_op,
+                                     stream,
+                                     debug_synchronous
+                                     ));
+
+    // End graph capture (since we can't malloc the temp storage inside the graph)
+    // and execute the graph (to get the temp storage size)
+    hipGraphExec_t graph_instance;
+    //graph_instance = test_utils::execGraphHelper(graph, stream);
```

Comment:
Is this some left-over debug code?

```suggestion
```

Reply:
Yes, that's left over debug code - sorry about that. I've cleaned it up now.